src/type_.rs

```diff
@@ -164,20 +181,28 @@ impl<'gcc, 'tcx> BaseTypeMethods<'tcx> for CodegenCx<'gcc, 'tcx> {
     fn type_kind(&self, typ: Type<'gcc>) -> TypeKind {
         if self.is_int_type_or_bool(typ) {
             TypeKind::Integer
-        } else if typ.is_compatible_with(self.float_type) {
-            TypeKind::Float
-        } else if typ.is_compatible_with(self.double_type) {
-            TypeKind::Double
-        } else if typ.is_vector() {
-            TypeKind::Vector
         } else if typ.get_pointee().is_some() {
             TypeKind::Pointer
+        } else if typ.is_vector() {
+            TypeKind::Vector
         } else if typ.dyncast_array().is_some() {
             TypeKind::Array
         } else if typ.is_struct().is_some() {
             TypeKind::Struct
         } else if typ.dyncast_function_ptr_type().is_some() {
             TypeKind::Function
+        } else if typ.is_compatible_with(self.float_type) {
+            TypeKind::Float
+        } else if typ.is_compatible_with(self.double_type) {
+            TypeKind::Double
+        } else if typ.is_floating_point() {
```

Comment:
You'll need to do this:
```suggestion
        }
        #[cfg(feature="master")]
        else if typ.is_floating_point() {
```

Reply:
This syntax isn't supported.
Unless you can put this above the `if`, I guess you can revert to 2 different implementations:
```
#[cfg(feature="master")]
if typ.is_floating_point() {
    match typ.get_size() {
                 2 => return TypeKind::Half,
                 4 => return TypeKind::Float,
                 8 => return TypeKind::Double,
                 16 => return TypeKind::FP128,
                 _ => return TypeKind::Void,
    }
}
```